docs/src/hugo/content/documentation/configuration.md

```diff
@@ -0,0 +1,1123 @@
+
+```
+nelson.cleanup.sweeper-delay = 24 hours
+```
+
+----
+
+## Database
+
+<span class="badge badge-warning">required</span>
+
+Nelson's H2 database requires some configuration to use. By default H2 will just write the database into the process-local folder, but this is typically not what you want. Most operators prefer the Nelson database to be on a redundant or backed-up known location.
+
+* [database.driver](#database-driver)
+* [database.connection](#database-connection)
+* [database.username](#database-username)
+* [database.password](#database-password)
+
+#### database.driver
+
+At the time of writing, Nelson only supported H2 as its backend datastore; this field should typically not be changed by administrators.
+
+```
+nelson.database.driver = "org.h2.Driver"
+```
+
+#### database.connection
+
+Depending on the driver specified in the `nelson.database.url` field, configure an appropriate JDBC string:
+
+```
+nelson.database.connection = "jdbc:h2:file:/opt/application/db/nelson;DATABASE_TO_UPPER=FALSE;AUTO_SERVER=TRUE;"
+```
+
+#### database.username
+
+If your database is using authentication, specify the username with the `username` field:
+
+```
+nelson.database.username = "admin"
+```
+
+#### database.password
+
+If your database is using authentication, specify the password with the `password` field:
+
+```
+nelson.database.password = "some password"
+```
+
+----
+
+## Datacenters
+
+<span class="badge badge-warning">required</span>
+
+The datacenters configuration section is what controls the various places that Nelson can deploy too. Within this section you can configure how Nelson knows how to talk to your scheudler of choice, instruct Nelson where it's Vault instance is, configure dynamic PKI and so forth. The structure of this section is namespaced by the datacenter in question; consider the following example:
+
+```
+datacenters {
+  texas {
+    ...
+  }
+  portland {
+    ...
+  }
+}
+```
+
+In this case two datacenters are configured: `texas` and `portland` - be aware that the names here are arbitrary, but once you choose them it is highly recomended to avoid changing them in the future as end-users can target specific datacenters for data-gravity or placement reason using the [datacenter stanza](/documentation/manifest.html#datacenters) of the Nelson manifest file, so future changes can be potentially disruptive. It is also highly adivsable that you choose unambiguous naming. For example, regions in [AWS](https://aws.amazon.com/) are things like <br/> `us-west-1` and `us-west-2` whilst region names in [GCP](https://cloud.google.com/) are `us-west1`; using these very similar names verbatim can be exceedingly confusing for users. Most companies internally have some taxonomy for their datacenter naming, so using something that users can identify with is usually advisable. Some Nelson users like to use regional airport codes, or even internal codewords - whatever scheme you choose, ensure users understand it.
```

Comment:
```suggestion
In this case two datacenters are configured: `texas` and `portland` - be aware that the names here are arbitrary, but once you choose them it is highly recommended to avoid changing them in the future as end-users can target specific datacenters for data-gravity or placement reason using the [datacenter stanza](/documentation/manifest.html#datacenters) of the Nelson manifest file, so future changes can be potentially disruptive. It is also highly advisable that you choose unambiguous naming. For example, regions in [AWS](https://aws.amazon.com/) are things like <br/> `us-west-1` and `us-west-2` whilst region names in [GCP](https://cloud.google.com/) are `us-west1`; using these very similar names verbatim can be exceedingly confusing for users. Most companies internally have some taxonomy for their datacenter naming, so using something that users can identify with is usually advisable. Some Nelson users like to use regional airport codes, or even internal codewords - whatever scheme you choose, ensure users understand it.
```